### ISSUE TITLE
Add support for running prepare on a specific ini file

### ISSUE THREAD

**Issue by mpkocher** [PacificBiosciences/pb-git#11]
It might be useful to make the `prepare` subcommand operate on a specific ini file.

`pb-git prepare pbcommand.ini`

**Comment by pb-cdunn** [comment]
I did consider this originally, but I thought it might cause confusion. It seems simpler and safer to expect a user to sync the git-commit for other modules before running `pb-git prepare`.

A more useful command could be something which delegates git commands to a subset of git-repos, running the same command in each of them. `git submodules` provides that, but of course we're not using that anymore. `gitslave` does that, via an easily installed perl script.. But really, a simple bash-script is enough for that. And in the future, I'd rather have support for "workspaces", which would somehow describe a subset of modules to work on locally, but the limiting factor is **mobs**.

What's the use-case?

**Comment by natechols** [comment]
A related use case: we'd like to be able to checkout only specific modules.

**Comment by pb-cdunn** [comment]
I don't think that `pb-git prepare` needs to operate on a subset, but yes, `pb-git checkout` could benefit.

This is pretty easy to do, but a good API is important. I generally avoid polymorphism in an API. In other words, if we want positional args to designate specific repos,

    pb-git checkout repo1 repo2 ...

then with no positional args,

    pb-git checkout

nothing should happen. I'm willing to relax that, but let's be clear about interaction with `--directory`. At the moment, we can do only one directory at a time. If you have repos into both pi and pivc, then you will need to call `pb-git checkout` at least twice. In that case, why not call it once for each repo?

    pb-git checkout --directory=pi --ini foo.ini
    pb-git checkout --directory=pivc --ini bar.ini

Alternatively, maybe we should call it just once. In that case, we could list all .ini on the command-line. But then we could eventually encounter a limit on the length of the command-line.

Maybe a better solution is to use a FOFN optionally.

    pb-git checkout --inis inis.fofn

That's ugly for a user, but fine for a script. Is that the use-case? What do you suggest?

And if we use a FOFN, then should relative ini paths be relative to CWD, or to `--directory`?

**Comment by natechols** [comment]
Yes, this is for a script (Jenkins builds).  I have no objection to using
an FOFN, I would assume that it would use CWD if --directory was not
specified.

**Comment by pb-cdunn** [comment]
pb-git --inis <(echo FALCON.ini pbsmrtpipe.ini) checkout

If you use full paths, then the checkout should occur in CWD, since `--directory` is `.` by default.

**Comment by mpkocher** [comment]
Thanks for the feature enhancements

**Comment by pb-cdunn** [comment]
```

**Comment by pb-cdunn** [comment]
Oops!

**Comment by pb-cdunn** [comment]
Fixed and submitted.
